kiliautoml/utils/download_assets.py

```diff
@@ -30,24 +30,28 @@ class DownloadedText:
     content: str
 
 
-ONE_MINUTE = 60
+DELAY = 60 / 250  # 250 calls per minutes
 
 
 @sleep_and_retry
-@limits(calls=250, period=ONE_MINUTE)
-def throttled_request(api_key, asset_content) -> Response:  # type: ignore
-    for _ in range(20):
-        try:
-            response = requests.get(
-                asset_content,
-                headers={
-                    "Authorization": f"X-API-Key: {api_key}",
-                },
-            )
-            assert response.status_code == 200
-            return response
-        except Exception:
-            time.sleep(1)
+@limits(calls=1, period=DELAY)
+def throttled_request(api_key, asset_content, use_header=True, k=0) -> Response:  # type: ignore
+    if k == 20:
+        raise Exception("Too many retries")
+    if use_header:
+        response = requests.get(
+            asset_content,
+            headers={
+                "Authorization": f"X-API-Key: {api_key}",
+            },
+        )
+    else:
+        response = requests.get(asset_content)
+    try:
+        assert response.status_code == 200
+        return response
+    except AssertionError:
+        return throttled_request(api_key, asset_content, use_header=not use_header, k=k + 1)
```

Comment:
why is there `not use_header`. If there is a valid justification, this should be added as a comment.

Reply:
Sometimes, the header breaks google bucket and just removing the header makes it work.